README.md

```diff
@@ -13,3 +13,139 @@ This is an NWB extension for storing metadata of devices used in optical experim
+        numerical_aperture : numeric, optional
+        core_diameter_in_um : numeric, optional
+    }
+    class ExcitationSource{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        illumination_type : text, optional
+        excitation_wavelength_in_nm : numeric, optional
+        power_in_W : numeric, optional
+        intensity_in_W_per_m2 : numeric, optional
+        exposure_time_in_s : numeric, optional
+    }
+    class PulsedExcitationSource{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        peak_power_in_W : numeric, optional
+        peak_pulse_energy_in_J : numeric, optional
+        pulse_rate_in_Hz : numeric, optional
+    }
+    class Photodetector{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        detector_type : text, optional
+        detected_wavelength_in_nm : numeric, optional
+        gain : numeric, optional
+    }
+    class DichroicMirror{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        cut_on_wavelength_in_nm : numeric, optional
+        cut_off_wavelength_in_nm : numeric, optional
+        reflection_bandwidth_in_nm : numeric, optional
+        transmission_bandwidth_in_nm : numeric, optional
+        angle_of_incidence_in_degrees : numeric, optional
+    }
+    class OpticalFilter{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        filter_type : text, optional
+    }
+    class BandOpticalFilter{
+        <<OpticalFilter>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        center_wavelength_in_nm : numeric
+        bandwidth_in_nm : numeric
+    }
+    class EdgeOpticalFilter{
+        <<OpticalFilter>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        cut_wavelength_in_nm : numeric
+        slope_in_percent_cut_wavelength : numeric, optional
+        slope_starting_transmission_in_percent : numeric, optional
+        slope_ending_transmission_in_percent : numeric, optional
+    }
+    class ObjectiveLens{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        numerical_aperture : numeric, optional
+        magnification : numeric, optional
+    }
+    class Microscope{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        microscopy_type : text, optional
```

Comment:
Can't this be determined indirectly from other fields? For inspection I'd recommend not having a loose classifier here when there's various ways it could be invalidated in complex ways by other fields cross-linked across objects

Reply:
That was my concern, too. On the one hand, I wanted to have a Device where you could specify the commercial or custom microscope as a whole and even have a field for the DOI referencing a specific implementation for custom setups (but now I realize that the `related_resources` field in the NWBFile metadata already serves this scope). On the other hand, if we use the same approach adopted in ndx-fiber-photometry, the ndx-microscopy will have a table (`MicroscopyTable`) where each row defines a specific configuration of the microscopy setup. Thus, the Microscope device will be redundant. I meant to discuss this in the meeting.

Reply:
I still recommend removing this classifier, more so now because there isn't a way to schematically restrict which type of microscopy it could be (including typos)

I also don't know if it applies to the `Microscope` or if the 'type' of the setup is something higher level. I'd expect such descriptions to be in things like the session or experiment description